src/apidocs/ApiDocsArea.tsx

```diff
@@ -0,0 +1,230 @@
+              {nameSuffix(kind, type)}
+            </Text>
+            {baseClasses && baseClasses.length > 0 && (
+              <BaseClasses value={baseClasses} />
+            )}
+            {docString && <DocString value={docString} />}
+          </>
+        )}
+      </Box>
+      {groupedChildren && groupedChildren.size > 0 && (
+        <Box pl={kind === "class" ? 2 : 0} mt={3}>
+          <Box
+            pl={kind === "class" ? 2 : 0}
+            borderLeftWidth={kind === "class" ? 1 : undefined}
+          >
+            {["function", "variable", "class"].map(
+              (childKind) =>
+                groupedChildren?.get(childKind as any) && (
+                  <Box mb={5} key={childKind}>
+                    <Text fontWeight="lg" mb={2}>
+                      {groupHeading(kind, childKind)}
+                    </Text>
+                    {groupedChildren?.get(childKind as any)?.map((c) => (
+                      <DocEntryNode key={c.id} docs={c} />
+                    ))}
+                  </Box>
+                )
+            )}
+          </Box>
+        </Box>
+      )}
+    </Box>
+  );
+};
+
+const groupHeading = (kind: string, childKind: string): string => {
+  switch (childKind) {
+    case "variable":
+      return "Fields";
+    case "class":
+      return "Classes";
+    case "function":
+      return kind === "class" ? "Methods" : "Functions";
+    default: {
+      throw new Error("Unexpected");
+    }
+  }
+};
+
+const formatName = (kind: string, fullName: string): string => {
+  // Add zero width spaces to allow breaking
+  return kind === "module"
+    ? fullName.replaceAll(/\./g, "\u200b.\u200b")
+    : fullName.split(".").slice(-1)[0];
+};
+
+const nameSuffix = (kind: string, type: string | undefined): string => {
+  if (kind === "function") {
+    return (type || "()").replace(/ -> None$/, "");
+  } else if (kind === "variable") {
+    return ": " + type;
+  }
+  return "";
+};
+
+const filterChildren = (
+  children: ApiDocsEntry[] | undefined
+): ApiDocsEntry[] | undefined =>
+  children
+    ? children.filter(
+        (c) => !(c.fullName.endsWith("__") && !c.fullName.endsWith("__init__"))
+      )
+    : undefined;
+
+function groupBy<T, U>(values: T[], fn: (x: T) => U): Map<U, T[]> {
+  const result = new Map<U, T[]>();
+  for (const v of values) {
+    const k = fn(v);
+    let array = result.get(k);
+    if (!array) {
+      array = [];
+      result.set(k, array);
+    }
+    array.push(v);
+  }
+  return result;
+}
+
+const BaseClasses = ({ value }: { value: ApiDocsBaseClass[] }) => {
+  const prefix = value.length === 1 ? "base class " : "base classes: ";
+  return (
+    <Text pl={2}>
+      {prefix}
+      {value.map((bc) => (
+        <a key={bc.fullName} href={"#" + bc.fullName}>
+          {bc.name}
+        </a>
+      ))}
+    </Text>
+  );
+};
+
+const DocString = ({ value }: { value: string }) => (
+  <Text fontSize="sm" mt={2} noOfLines={2} fontWeight="normal">
```

Comment:
noOfLines is just a hack, we need to write good first lines of docstrings and then extract just those to show here